lib/puppet/resource.rb

```diff
@@ -633,6 +561,12 @@ def validate_parameter(name)
     raise Puppet::ParseError.new(_("no parameter named '%{name}'") % { name: name }, file, line) unless valid_parameter?(name)
   end
 
+  # This method, together with #file and #line, makes it possible for a Resource to be a 'source_pos' in a reported issue.
+  # @return [Integer] Instances of this class will always return 0.
+  def pos
+    0
+  end
```

Comment:
This means it will report a position - no? As opposed to `pos` being `nil` with the meaning that the position is unknown...